csrc/python_frontend/python_bindings.cpp

```diff
@@ -2911,8 +2911,71 @@ void initNvFuserPythonBindings(PyObject* module) {
       py::arg("correction") = 1,
       py::arg("keepdim") = false,
       py::return_value_policy::reference);
-  //! The SchedOperators class is a nested class of FusionDefinition to allow
-  //! the user to query the class for the list of schedule operators.
+  nvf_ops.def(
+      "sdpfa_fwd",
+      [](FusionDefinition::Operators& self,
+         Tensor query,
+         Tensor key,
+         Tensor value,
+         std::optional<Scalar> dropout_p,
+         std::optional<Scalar> is_causal,
+         std::optional<Scalar> scale) -> decltype(auto) {
+        FUSER_PERF_SCOPE("Operators.sdpfa_fwd");
+        NVF_CHECK(
+            self.validUse(), "Attempting to add to a completed definition!");
+        FusionDefinition* fd = self.fusion_definition;
+        size_t ndims = query.dims;
+        Tensor output = fd->defineTensor(ndims);
+        Tensor log_sumexp = fd->defineTensor(ndims - 1);
+        Tensor query_seq_len = fd->defineTensor(0);
+        Tensor key_seq_len = fd->defineTensor(0);
+        Tensor philox_seed = fd->defineTensor(0);
+        Tensor philox_offset = fd->defineTensor(0);
+        Tensor debug_attn_mask = fd->defineTensor(0);
+
+        auto dropout_p_state = dropout_p.has_value()
+            ? fd->recordingState(dropout_p.value()())
+            : State(0, serde::StateType::None);
+        auto is_causal_state = is_causal.has_value()
+            ? fd->recordingState(is_causal.value()())
+            : State(0, serde::StateType::None);
+        auto scale_state = scale.has_value()
+            ? fd->recordingState(scale.value()())
+            : State(0, serde::StateType::None);
```

Comment:
Is default 0 the right behavior?

A quick dive into the sdpa_fwd, looks like a none scale triggers a different code path vs some concrete value.

Reply:
That is the state index, and not the default value.
I call the C++ API using `nullptr`.

Reply:
:face_with_head_bandage: 

Of course. Sorry about misreading this.....